tests/test_ecu_variant_matching.py

```diff
@@ -105,6 +117,17 @@ def supplier_service(monkeypatch: pytest.MonkeyPatch, dummy_response: Response)
         description=None,
         semantic=None,
         admin_data=None,
+        protocol_snrefs=[],
+        related_diag_comm_refs_raw=[],
+        diagnostic_class=None,
+        is_mandatory_raw=None,
+        is_executable_raw=None,
+        is_final_raw=None,
+        comparam_refs=NamedItemList[CommunicationParameterRef](),
```

Comment:
refs as named list?

Reply:
In contrast to the XSD, `CommunicationParameterRef` is not derived from `OdxLinkRef`. This mainly because it contains plenty of additional information, such as the value for the respective communication parameter but also because `OdxLinkRef` is currently a frozen dataclass in order for it to be hashable. (its `short_name` property currently is a forward to the short name of the referenced comparam spec: https://github.com/mercedes-benz/odxtools/blob/main/odxtools/communicationparameterref.py#L135 ) I don't have an elegant solution here, and I'm not even sure if one exists...

Reply:
My suggestion in this case, is to use a different suffix, for example `Reference` instead of `Ref`, so that the two types don't get mixed up

Reply:
Okay: I have refactored the comparam classes, but it seems like the comparam stuff in the ODX spec is quite quirky, so I decided to deviate from the ODX specification's structure a bit: I renamed the "*Comparam" to "*ComparamSpec" and "CommunicationParameterRef" to "Comparam", moved the physical default value for complex comparam specs into the class itself instead of keeping it weirdly external as mandated by ODX, plus  a few other things.

since this a rather big endeavor, I'd like to do this in a later pull request...